panel/io/server.py

```diff
@@ -485,27 +485,28 @@ async def get(self, *args, **kwargs):
                 token = session.token
             logger.info(LOG_SESSION_CREATED, id(session.document))
             with set_curdoc(session.document):
-                import inspect
-                authorize_callback_signature = inspect.signature(config.authorize_callback)
-                authorize_callback_parameters = authorize_callback_signature.parameters
-                authorized = False
-                if len(authorize_callback_parameters) == 1:
-                    authorized = config.authorize_callback(state.user_info)
-                if len(authorize_callback_parameters) == 2:
-                    authorized = config.authorize_callback(state.user_info, self.request.path)
-                if config.authorize_callback and not authorized:
-                    if config.auth_template:
-                        with open(config.auth_template) as f:
-                            template = _env.from_string(f.read())
-                    else:
-                        template = ERROR_TEMPLATE
-                    page = template.render(
-                        npm_cdn=config.npm_cdn,
-                        title='Panel: Authorization Error',
-                        error_type='Authorization Error',
-                        error='User is not authorized.',
-                        error_msg=f'{state.user} is not authorized to access this application.'
-                    )
+                if config.authorize_callback:
+                    import inspect
+                    authorize_callback_signature = inspect.signature(config.authorize_callback)
+                    authorize_callback_parameters = authorize_callback_signature.parameters
+                    authorized = False
+                    if len(authorize_callback_parameters) == 1:
+                        authorized = config.authorize_callback(state.user_info)
+                    if len(authorize_callback_parameters) == 2:
+                        authorized = config.authorize_callback(state.user_info, self.request.path)
+                    if not authorized:
```

Comment:
This branching doesn't look right, if `authorized` is true it won't hit the else branch below.

Reply:
yup, will fix...was attempting to juggle too many things atm

Reply:
I can move the creation of the `page` object above the `if config.authorize_callback`, but this always assumes the request and requester are authorized. If a user supplies a `config.authorize_callback` and they are authorized, then the initial `page` creation is returned, however, if they are not authorized then the orginal `page` object is updated to be the `ERROR_TEMPLATE` object.

I'll make a new commit to see if you think this is okay or not. If not, then I'm happy to write it as a `try/except/finally` as you suggested above.

Reply:
Yeah, this is still not correct. I'll push a fix.